### PR TITLE
[WFLY-10178] Support for MR-JAR builds & README -> AsciiDoc

### PR DESCRIPTION
**Pull request by dmlloyd**
https://issues.jboss.org/browse/WFLY-10178
https://github.com/wildfly/wildfly-proposals/pull/46

### CODE REVIEW

**Comment by dmlloyd** [comment]
This was tested with wildfly-common, jboss-logmanager, and jboss-modules.

File: pom.xml [hunk 1]
```diff
@@ -77,7 +77,8 @@
     <version.clean.plugin>3.1.0</version.clean.plugin>
     <version.clover.plugin>4.1.2</version.clover.plugin>
     <version.cobertura.plugin>2.7</version.cobertura.plugin>
-    <version.compiler.plugin>3.7.0</version.compiler.plugin>
+    <!-- As of this writing, the upstream plugin does not support additionalClassPathElements -->
+    <version.compiler.plugin>3.7.0-jboss-1</version.compiler.plugin>
```
[the remark this answers is not in the document]
Comment (dmlloyd) [reply pom.xml]: I have published a 3.8.0-jboss-1 as well.  But I don't think we should make the move from 3.8.0 at the same time as this change.  That can be a separate PR.

**Comment by ctomc** [comment]
Looks good, only question is, should JDK 9 and 10 we be even considered as they are both EOL
Having JDK8 and 11 only should be just fine.

**Comment by dmlloyd** [comment]
Yes, I think so. I think there's no reason to go out of our way to unsupport middle versions based on the support policy of a completely different company.

Once our minimum version requirements are ubiquitously higher, we can begin to roll off older version support, as we did for 6 and 7.

**Comment by rsvoboda** [comment]
I think many MR jars put stuff under `META-INF/versions/9` so we should probably keep profiles for 9 and 10.

Or should we move on and use just `META-INF/versions/11` ?

**Comment by dmlloyd** [comment]
Since this POM is used by community projects, I think supporting 8 through 12 is appropriate for now.

**Comment by dmlloyd** [comment]
If nobody else has any comments, I'd like to get this merged.  I have downstream projects that are waiting for this support.

File: pom.xml [hunk 2]
```diff
@@ -841,6 +844,391 @@
...
+            <executions>
+              <execution>
+                <id>java11-test</id>
+                <phase>test</phase>
+                <goals>
+                  <goal>test</goal>
+                </goals>
+                <configuration>
+                  <jvm>${java11.home}/bin/java</jvm>
+                  <classesDirectory>${project.build.directory}/classes/META-INF/versions/10</classesDirectory>
```
Comment (jamezp) [inline pom.xml]: Should his be `12` with `11` added below in the `additionalClasspathElement`?

**Comment by dmlloyd** [comment]
Hold on, I have some newer changes...

**Comment by dmlloyd** [comment]
OK this is revised with the new code.  Sorry about the confusion.  Please review _this_ version and in the meantime I'll run it through additional testing to make sure nothing new has broken.

**Comment by dmlloyd** [comment]
Testing seems OK; tested the MR JAR build on wildfly-common and the standard parent on wildfly-core.

File: pom.xml [hunk 3]
```diff
@@ -1,859 +1,86 @@
...
-      <name>Public Domain</name>
-      <url>http://repository.jboss.org/licenses/cc0-1.0.txt</url>
-      <distribution>repo</distribution>
-    </license>
-  </licenses>
-
-  <organization>
-    <name>JBoss by Red Hat</name>
+    <name>JBoss Parent Parent POM</name>
+    <description>Parent Parent POM for JBoss projects. Provides default project build configuration.</description>
```
Comment (bstansberry) [inline pom.xml]: Perhaps a bit more words about the structure? For example

"Provides, via child modules, default project build configurations, both for projects producing non-compiled artifacts and traditional ones compiled using a single JDK version, and also for projects producing multi-release jars with resources compiled using multiple JDK versions."

Comment (dmlloyd) [reply pom.xml]: Sure, I was pretty lazy on this one :)

File: pom.xml (hunk 2 above)
Comment (dmlloyd) [reply pom.xml]: This is fixed now.

File: pom.xml (hunk 3 above)
Comment (dmlloyd) [reply pom.xml]: Fixed.

**Comment by dmlloyd** [comment]
OK I'll merge this in a few minutes unless another objection arises imminently.

File: pom.xml (hunk 3 above)
Comment (bstansberry) [reply pom.xml]: Well, I think my 32 added words is a smaller contribution than yours! ;)

**Pull request merged**